esque/clients.py

```diff
@@ -40,27 +75,105 @@ def consume_ping(self) -> Optional[Tuple[str, int]]:
         delta_sent = pendulum.now() - msg_sent_at
         return msg.key(), delta_sent.microseconds / 1000
 
-    def _assign_exact_partitions(self, topic: str) -> None:
-        self._consumer.assign([TopicPartition(topic=topic, partition=0, offset=0)])
 
+class FileConsumer(AbstractConsumer):
+    def __init__(self, group_id: str, topic_name: str, working_dir: pathlib.Path, last: bool):
+        super().__init__(group_id, topic_name, last)
+        self.working_dir = working_dir
+        offset_reset = "earliest"
+        if last:
+            offset_reset = "latest"
+
+        self._config.update({"default.topic.config": {"auto.offset.reset": offset_reset}})
+        self._consumer = confluent_kafka.Consumer(self._config)
+        self._subscribe(topic_name)
```

Comment:
> Consider for example non-default partitioning where a producer decides for itself which partition to produce the message to.
Also in case of AVRO, the schema id can change at different offsets within the partitions. Your AvroFileWriter can't handle that.

Actually it does, if I understood you correctly. I store for each serialized message the related schema_directory. And for each message it the schema ids.